cli/cmd/install.go

```diff
@@ -205,56 +219,155 @@ func validateAndBuildConfig(options *installOptions) (*installConfig, error) {
+
+	return InjectYAML(&injectTmpl, w, ioutil.Discard, injectOptions)
+}
+
+func loadChart(config *installConfig) (*chart.Chart, error) {
+	chrt := &chart.Chart{}
+
+	// Load chart metadata
+	if err := loadMetaData(chrt); err != nil {
+		return &chart.Chart{}, err
+	}
+
+	// Always load the base template
+	if err := loadTemplate(chrt, baseTemplateName, baseTemplatePath); err != nil {
+		return &chart.Chart{}, err
+	}
+
 	if config.EnableTLS {
-		tlsTemplate, err := template.New("linkerd").Parse(install.TLSTemplate)
-		if err != nil {
-			return err
-		}
-		err = tlsTemplate.Execute(buf, config)
-		if err != nil {
-			return err
+		if err := loadTemplate(chrt, tlsTemplateName, tlsTemplatePath); err != nil {
+			return &chart.Chart{}, err
 		}
 
 		if config.ProxyAutoInjectEnabled {
-			proxyInjectorTemplate, err := template.New("linkerd").Parse(install.ProxyInjectorTemplate)
-			if err != nil {
-				return err
-			}
-			err = proxyInjectorTemplate.Execute(buf, config)
-			if err != nil {
-				return err
+			if err := loadTemplate(chrt, proxyInjectorName, proxyInjectorTemplatePath); err != nil {
+				return &chart.Chart{}, err
 			}
 		}
 	}
 
-	injectOptions := newInjectOptions()
-	injectOptions.proxyConfigOptions = options.proxyConfigOptions
+	return chrt, nil
+}
 
-	// Special case for linkerd-proxy running in the Prometheus pod.
-	injectOptions.proxyOutboundCapacity[config.PrometheusImage] = prometheusProxyOutboundCapacity
+func loadMetaData(chrt *chart.Chart) error {
+	metadataBytes, err := ioutil.ReadFile("../install/Chart.yaml")
+	if err != nil {
+		return err
+	}
+	metadata, err := chartutil.UnmarshalChartfile(metadataBytes)
+	if err != nil {
+		return err
+	}
+	chrt.Metadata = metadata
 
-	return InjectYAML(buf, w, ioutil.Discard, injectOptions)
+	return nil
 }
 
-func (options *installOptions) validate() error {
-	if _, err := log.ParseLevel(options.controllerLogLevel); err != nil {
-		return fmt.Errorf("--controller-log-level must be one of: panic, fatal, error, warn, info, debug")
+func loadTemplate(chrt *chart.Chart, tmplName string, tmplPath string) error {
+	tmpl, err := ioutil.ReadFile(tmplPath)
+	if err != nil {
+		return err
 	}
+	chrt.Templates = append(chrt.Templates, &chart.Template{Name: tmplName, Data: tmpl})
 
-	if options.proxyAutoInject && options.singleNamespace {
-		return fmt.Errorf("The --proxy-auto-inject and --single-namespace flags cannot both be specified together")
+	return nil
+}
+
+func renderRawValues(chrt *chart.Chart, config *installConfig) (chartutil.Values, error) {
+	// Initialize helper structs for use in rendering values
+	caps := &chartutil.Capabilities{
+		APIVersions:   chartutil.DefaultVersionSet,
+		KubeVersion:   chartutil.DefaultKubeVersion,
+		TillerVersion: tversion.GetVersionProto(),
 	}
 
-	return options.proxyConfigOptions.validate()
+	renderOpts := renderutil.Options{
+		ReleaseOptions: chartutil.ReleaseOptions{
+			Name:      "release-name",
```

Comment:
We'll want this to be `linkerd` (and then include the release name in the chart as that is the best practice standard).